airflow/providers/common/sql/hooks/sql.py

```diff
@@ -207,6 +219,24 @@ def get_uri(self) -> str:
         conn.schema = self.__schema or conn.schema
         return conn.get_uri()
 
+    def get_sqlalchemy_url(self) -> URL:
+        """
+        Return a Sqlalchemy.engine.URL object from the connection.
+
+        :return: the extracted sqlalchemy.engine.URL object.
+        """
+        conn = self.get_connection(getattr(self, self.conn_name_attr))
+        conn.schema = self.__schema or conn.schema
+        conn.port = conn.port or 5432
+        return URL.create(
+            drivername=self.drivername,
+            username=conn.login,
+            password=conn.password,
+            host=conn.host,
+            port=conn.port,
+            database=conn.schema,
+        )
```

Comment:
That is also does not work in all cases. Recent example is AthenaSQLHook

https://github.com/apache/airflow/blob/666127238b727c9ee5dfaa1d65782b9d1ab105ed/airflow/providers/amazon/aws/hooks/athena_sql.py#L142-L155

Reply:
I understand. I've updated the PR to not have any implementation so that this kind of custom implementations can be done in the providers.